test/unit/test_type.c

```diff
@@ -512,6 +512,185 @@ static bool test_type_as_string(void) {
+	free(pretty_str);
+	pretty_str = rz_type_as_pretty_string(typedb, ttype, NULL, RZ_TYPE_PRINT_UNFOLD_ANONYMOUS_ONLY, -1);
+	mu_assert_streq(pretty_str, pretty_struct_in_struct, "struct in struct type anon unfold is ugly");
+	free(pretty_str);
+	free(ttype);
+
+	error_msg = NULL;
+	ttype = rz_type_parse_string_single(typedb->parser, pretty_union_of_struct, &error_msg);
+	mu_assert_notnull(ttype, "union of struct type parse unsuccessfull");
+	mu_assert_null(error_msg, "parsing errors");
+	pretty_str = rz_type_as_pretty_string(typedb, ttype, NULL, RZ_TYPE_PRINT_UNFOLD_ANONYMOUS_ONLY, 2);
+	mu_assert_streq(pretty_str, pretty_union_of_struct, "union of struct type is ugly");
+	free(pretty_str);
+	pretty_str = rz_type_as_pretty_string(typedb, ttype, "mult1", RZ_TYPE_PRINT_MULTILINE, 2);
+	mu_assert_streq(pretty_str, pretty_union_of_struct_multiline1, "union of struct type multiline 1 is ugly");
+	free(pretty_str);
+	pretty_str = rz_type_as_pretty_string(typedb, ttype, "anonmult", RZ_TYPE_PRINT_MULTILINE | RZ_TYPE_PRINT_UNFOLD_ANONYMOUS_ONLY, 10);
+	mu_assert_streq(pretty_str, pretty_union_of_struct_anon_multiline, "union of struct type anon multiline is ugly");
+	free(pretty_str);
+	pretty_str = rz_type_as_pretty_string(typedb, ttype, "maxmult", RZ_TYPE_PRINT_MULTILINE, -3);
+	mu_assert_streq(pretty_str, pretty_union_of_struct_max_multiline, "union of struct type max multiline is ugly");
+	free(pretty_str);
+	free(ttype);
+
+	error_msg = NULL;
+	ttype = rz_type_parse_string_single(typedb->parser, pretty_enum, &error_msg);
+	mu_assert_notnull(ttype, "enum type parse unsuccessfull");
+	mu_assert_null(error_msg, "parsing errors");
+	pretty_str = rz_type_as_pretty_string(typedb, ttype, NULL, RZ_TYPE_PRINT_NO_OPTS, 10);
+	mu_assert_streq(pretty_str, pretty_enum, "enum type is ugly");
+	free(pretty_str);
+	pretty_str = rz_type_as_pretty_string(typedb, ttype, "enumult", RZ_TYPE_PRINT_MULTILINE, -2);
+	mu_assert_streq(pretty_str, pretty_enum_multiline, "enum type multiline is ugly");
+	free(pretty_str);
+	free(ttype);
+
+	mu_end;
+}
+
+static bool test_temp(void) {
+	RzTypeDB *typedb = rz_type_db_new();
+	mu_assert_notnull(typedb, "Couldn't create new RzTypeDB");
+	mu_assert_notnull(typedb->types, "Couldn't create new types hashtable");
+	const char *dir_prefix = rz_sys_prefix(NULL);
+	rz_type_db_init(typedb, dir_prefix, "x86", 64, "linux");
+
+	char *error_msg = NULL;
+	RzType *ttype = rz_type_parse_string_single(typedb->parser, pretty_complex_const_pointer, &error_msg);
+	mu_assert_notnull(ttype, "struct array ptr func ptr type parse unsuccessfull");
+	mu_assert_null(error_msg, "parsing errors");
+
+	error_msg = NULL;
+	ttype = rz_type_parse_string_single(typedb->parser, pretty_complex_const_pointer, &error_msg);
+	mu_assert_notnull(ttype, "struct array ptr func ptr type parse unsuccessfull");
+	mu_assert_null(error_msg, "parsing errors");
+
+	mu_end;
+}
```

Comment:
is this here on purpose?

Reply:
Oh, sorry. It was just a temporary test :)

Reply:
> Right @XVilka ? I think that would be the proof that everything works well ;)

Agreed! But should that be done in this PR, or should I be creating another PR for that?

Reply:
I think this PR should be fine because if all works well it should be just a matter of removing a lot of the existing code and replace it with a call to `rz_type_as_pretty_string` with the right parameters. If it does not work that way, it means you probably have to adjust the API anyway, i think.

Reply:
I have refactored the other type print functions to use this API. I had to add a few more pretty printing options (though it was very easy to add them) so that the expected output of the original functions stays the same. I will do the same for `tc` once you review the refactored functions. Thank you.